### PR TITLE
Add new workload Microsoft.Exchange.Rbac to beta endpoint

### PR DESCRIPTION
**Pull request by cubika**
Add new workload "Microsoft.Exchange.Rbac" for Exchange provider("/roleManagement/exchange") with a bunch of existing rbac related entities.

Schema PR # 7138174 , cc @Jarbas-MSFT

### CODE REVIEW

**Comment by FaithOmbongi** [comment]
Hi @Jarbas-MSFT - I'll assign this to you coz it comes through an E+D PM owner. When you get time, please review? Thanks.

cc @Lauragra for visibility.

**Comment by FaithOmbongi** [comment]
Hi @Jarbas-MSFT - I'll assign this you coz it comes through an E+D PM owner.
@cubika - Please update the description to add the schema PR number or the roadmap item number that relate to these changes. Jarbas will need it for content review.

cc @Lauragra for visibility.

File: api-reference/beta/resources/unifiedrbacapplication.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Represents a role management container for unified role definitions and role assignments for role-based access control (RBAC) providers in Microsoft 365. This is a shared entity meant to replace [rbacApplication](./rbacapplication.md). Currently only Exchange RBAC applications are supported.
+
+## Methods
+
+| Method       | Return Type | Description |
+|:-------------|:------------|:------------|
+| [Create unifiedRoleAssignment](../api/rbacapplication-post-roleassignments.md) | [unifiedRoleAssignment](unifiedroleassignment.md) | Create a new [unifiedRoleAssignment](../resources/unifiedroleassignment.md) object. |
+| [List roleAssignments](../api/rbacapplication-list-roleassignments.md) | [unifiedRoleAssignment](unifiedroleassignment.md) collection | Get a list of [unifiedRoleAssignment](../resources/unifiedroleassignment.md) objects for an RBAC provider. You can only query specific instances by filtering on **roleDefinitionId** or **principalId**. |
```
Comment (Jarbas-MSFT) [inline api-reference/beta/resources/unifiedrbacapplication.md:23]: According to the [List roleAssignments](https://raw.githubusercontent.com/microsoftgraph/microsoft-graph-docs/main/api-reference/beta/api/rbacapplication-list-roleassignments.md
) API topic, you can also filter on **appScopeId**. If this is relevant, please also include **appScopeId** as a supported filtering option.

File: changelog/Microsoft.Exchange.Rbac.json [hunk 2]
```diff
@@ -0,0 +1,86 @@
+{
```
Comment (Jarbas-MSFT) [inline changelog/Microsoft.Exchange.Rbac.json:1]: Based on previous changelog entries for similar changes, the correct changelog file to publish these changes is: "changelog/Microsoft.DirectoryServices.json". Please move these changes to the correct changelog file.

The following resources have already been published and you don't need to refer them here again. Please remove the already published resources from this changelog record.
- unifiedRolePermission
- appScope
- directoryObject
- roleManagement
- unifiedRbacResourceAction
- unifiedRbacResourceNamespace
- unifiedRoleAssignment
- unifiedRoleDefinition

Also, this changelog record is missing an entry for the addition of the exchange relationship to the roleManagement resource. Please add the missing entry to the changelog.

File: api-reference/beta/resources/rolemanagement.md [hunk 3]
```diff
@@ -20,12 +20,14 @@ The unified role management API currently supports the following RBAC providers
...
 - directory (Azure AD directory roles)
 - entitlement management (Azure AD entitlement management)
+- Exchange
  
 For more information, see: 
 * [Roles in Microsoft 365, including Azure AD, service-specific and cross-service roles](/azure/active-directory/roles/concept-understand-roles#how-azure-ad-roles-are-different-from-other-microsoft-365-roles) 
 * [Administrator role permissions in Azure Active Directory](/azure/active-directory/users-groups-roles/directory-assign-admin-roles).
 * [Delegation and roles in Azure AD entitlement management](/azure/active-directory/governance/entitlement-management-delegate).
 * [Role-based access control (RBAC) with Microsoft Intune](/mem/intune/fundamentals/role-based-access-control)
+* [Understanding Role Based Access Control in Exchange Online](/exchange/understanding-role-based-access-control-exchange-2013-help)
```
[the remark this answers is not in the document]
Comment (cubika) [reply api-reference/beta/resources/rolemanagement.md:30]: Yes, the article applies to Exchange Online.

File: api-reference/beta/resources/unifiedrbacapplication.md (hunk 1 above)
Comment (cubika) [reply api-reference/beta/resources/unifiedrbacapplication.md]: Updated.

File: api-reference/beta/resources/rolemanagement.md [hunk 4]
```diff
@@ -20,12 +20,14 @@ The unified role management API currently supports the following RBAC providers
 - device management (Intune)
 - directory (Azure AD directory roles)
 - entitlement management (Azure AD entitlement management)
+- Exchange
```
[the remark this answers is not in the document]
Comment (cubika) [reply api-reference/beta/resources/rolemanagement.md]: Added. Yes, the new API set is for Exchange Online.

**Comment by cubika** [comment]
Added some sections about the permissions used for the new provider.

File: api-reference/beta/api/unifiedroledefinition-get.md [hunk 5]
```diff
@@ -57,6 +58,14 @@ Depending on the RBAC provider and the permission type (delegated or application
 |Delegated (personal Microsoft account) | Not supported.    |
 |Application | EntitlementManagement.Read.All, EntitlementManagement.ReadWrite.All |
 
+### For an Exchange Online provider
+
+|Permission type      | Permissions (from least to most privileged)              |
+|:--------------------|:---------------------------------------------------------|
+|Delegated (work or school account) |  RoleManagement.Read.Exchange, RoleManagement.Read.All, RoleManagement.ReadWrite.Exchange   |
```
[the remark this answers is not in the document]
Comment (cubika) [reply api-reference/beta/api/unifiedroledefinition-get.md:65]: Thanks for the reminding. Yes, these are new permissions. I've updated in latest iteration.

**Comment by JarbasHorst** [comment]
Hi @Lauragra, this PR is ready for final review.

**Review by Lauragra — APPROVED** [review]

File: api-reference/beta/resources/unifiedrbacapplication.md
[the remark this answers is not in the document]
Comment (cubika) [reply api-reference/beta/resources/unifiedrbacapplication.md:1]: The new entity is designed by AAD Rbac team to replace existing rbacApplication which has some issues. Copy the reason from API review doc:

> rbacApplication was modeled as a shared entity, but it was not implemented as such in MSGraph. This broke the critical validation rule forbidding containment to foreign entities. Over time, it was extended by multiple workloads such as PIM, Access Reviews etc. (so we can't make it a SharedEntity without breaking changes) adding workload specific properties that other workloads did not implement/do not understand.
This has created unintended broken API routes, leading to inconsistent workload routing.

File: api-reference/beta/resources/rolemanagement.md [hunk 6]
```diff
@@ -43,6 +45,7 @@ None.
 |deviceManagement|[rbacApplicationMultiple](rbacapplicationmultiple.md)| Provides access to role definitions and role assignments of an Intune RBAC provider. Read-only. Nullable.|
 |directory|[rbacApplication](rbacapplication.md)|Provides access to role definitions and role assignments of an Azure AD RBAC provider. Read-only. Nullable.|
 |entitlementManagement|[rbacApplication](rbacapplication.md)| Provides access to role definitions and role assignments of Azure AD entitlement management. Read-only. Nullable.|
+|exchange|[unifiedRbacApplication](unifiedrbacapplication.md)| Provides access to role definitions and role assignments of Exchange providers. Read-only. Nullable.|
```
[the remark this answers is not in the document]
Comment (cubika) [reply api-reference/beta/resources/rolemanagement.md:48]: Same reason as above. https://microsoftgraph.visualstudio.com/onboarding/_git/onboarding/pullrequest/11887 has more details.

File: api-reference/beta/api/rbacapplication-list-roleassignments.md [hunk 7]
```diff
@@ -40,6 +41,14 @@ Depending on the RBAC provider and the permission type (delegated or application
 |Delegated (personal Microsoft account) | Not supported.    |
 |Application | Not supported. |
 
+### For an Exchange Online provider
```
Comment (Lauragra) [inline api-reference/beta/api/rbacapplication-list-roleassignments.md:44]: One bit of feedback based on the fact that we'll be automating the generation of the permissions tables in the near future. When we add another permissions table, we also need an HTTP request code block in the HTTP request section. Please add an additional code block that shows the REST syntax to each topic that includes the additional permissions table.

File: api-reference/beta/api/rbacapplication-list-roleassignments.md [hunk 8]
```diff
@@ -40,6 +41,14 @@ Depending on the RBAC provider and the permission type (delegated or application
...
+
+|Permission type      | Permissions (from least to most privileged)              |
+|:--------------------|:---------------------------------------------------------|
+|Delegated (work or school account) |  RoleManagement.Read.Exchange, RoleManagement.Read.All, RoleManagement.ReadWrite.Exchange   |
+|Delegated (personal Microsoft account) | Not supported.    |
+|Application | Not supported. |
+
 
 ## HTTP request
```
Comment (Lauragra) [inline api-reference/beta/api/rbacapplication-list-roleassignments.md:53]: We're missing the HTTP request section for the Exchange Online provider in this topic, it looks like.

File: api-reference/beta/resources/rolemanagement.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (psignoret) [reply api-reference/beta/resources/rolemanagement.md:48]: Gotcha, thanks for the details!

**Review by Lauragra — APPROVED** [review]

**Pull request merged**